zebra-consensus/src/transaction.rs

```diff
@@ -345,6 +344,34 @@ where
 
             tracing::trace!(?tx_id, "passed quick checks");
 
+            if let Some(block_time) = req.block_time() {
+                check::lock_time_has_passed(&tx, req.height(), block_time)?;
+            } else {
+                // This state query is much faster than loading UTXOs from the database,
+                // so it doesn't need to be executed in parallel
+                let state = state.clone();
+                let next_median_time_past = Self::mempool_best_chain_next_median_time_past(state).await?;
```

Comment:
Optional: skip this request if the `lock_time` is a height.

Reply:
I'll make this change in PR #6031.

Reply:
Fixed in 8050557681 in PR #6031